pkg/hintrunner/hint.go

```diff
@@ -371,13 +373,383 @@ func (hint SquareRoot) Execute(vm *VM.VirtualMachine) error {
+	if err != nil {
+		return fmt.Errorf("read initialized dicts: %w", err)
+	}
+
+	// find for the segment info pointer
+	segmentInfoOffset, overflow := safemath.SafeOffset(arenaPtr.Offset, -3)
+	if overflow {
+		return fmt.Errorf("look for segment info pointer: overflow: %s - 3", arenaPtr)
+	}
+	segmentInfoMv, err := vm.Memory.Read(arenaPtr.SegmentIndex, segmentInfoOffset)
+	if err != nil {
+		return fmt.Errorf("read segment info pointer: %w", err)
+	}
+	segmentInfoPtr, err := segmentInfoMv.MemoryAddress()
+	if err != nil {
+		return fmt.Errorf("expected pointer to segment info but got a felt: %w", err)
+	}
+
+	// with the segment info pointer and the number of initialized dictionaries we know
+	// where to write the new dictionary
+	newDictAddress := ctx.DictionaryManager.NewDictionary(vm)
+	mv := mem.MemoryValueFromMemoryAddress(&newDictAddress)
+	insertOffset := segmentInfoPtr.Offset + initializedDicts*3
+	if err = vm.Memory.Write(segmentInfoPtr.SegmentIndex, insertOffset, &mv); err != nil {
+		return fmt.Errorf("write new dictionary to segment info: %w", err)
+	}
+	return nil
+}
+
+type Felt252DictEntryInit struct {
+	DictPtr ResOperander
+	Key     ResOperander
+}
+
+func (hint Felt252DictEntryInit) String() string {
+	return "Felt252DictEntryInit"
+}
+
+func (hint *Felt252DictEntryInit) Execute(vm *VM.VirtualMachine, ctx *HintRunnerContext) error {
+	dictPtr, err := ResolveAsAddress(vm, hint.DictPtr)
+	if err != nil {
+		return fmt.Errorf("resolve dictionary pointer: %w", err)
+	}
+
+	key, err := ResolveAsFelt(vm, hint.Key)
+	if err != nil {
+		return fmt.Errorf("resolve key: %w", err)
+	}
+
+	prevValue, err := ctx.DictionaryManager.At(&dictPtr, &key)
+	if err != nil {
+		return fmt.Errorf("get dictionary entry: %w", err)
+	}
+	if prevValue == nil {
+		mv := mem.EmptyMemoryValueAsFelt()
+		prevValue = &mv
+		_ = ctx.DictionaryManager.Set(&dictPtr, &key, prevValue)
+	}
+	return vm.Memory.Write(dictPtr.SegmentIndex, dictPtr.Offset+1, prevValue)
+}
+
+type Felt252DictEntryUpdate struct {
+	DictPtr ResOperander
+	Value   ResOperander
+}
+
+func (hint Felt252DictEntryUpdate) String() string {
+	return "Felt252DictEntryUpdate"
+}
+
+func (hint *Felt252DictEntryUpdate) Execute(vm *VM.VirtualMachine, ctx *HintRunnerContext) error {
+	dictPtr, err := ResolveAsAddress(vm, hint.DictPtr)
+	if err != nil {
+		return fmt.Errorf("resolve dictionary pointer: %w", err)
+	}
+
+	keyPtr, err := dictPtr.AddOffset(-3)
+	if err != nil {
+		return fmt.Errorf("get key pointer: %w", err)
+	}
+	keyMv, err := vm.Memory.ReadFromAddress(&keyPtr)
+	if err != nil {
+		return fmt.Errorf("read key pointer: %w", err)
+	}
+	key, err := keyMv.FieldElement()
+	if err != nil {
+		return fmt.Errorf("expected key to be a field element: %w", err)
+	}
+
+	value, err := hint.Value.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve value: %w", err)
+	}
+
+	return ctx.DictionaryManager.Set(&dictPtr, key, &value)
+}
+
+type GetSegmentArenaIndex struct {
```

Comment:
Why is this struct the only one that's `String()` value is different than struct name?

Reply:
oops!